charger/delta.go

```diff
@@ -156,44 +207,50 @@ func (wb *Delta) Status() (api.ChargeStatus, error) {
 	// 7: Suspended EVSE
 	// 8: Not ready
 	// 9: Faulted
+
 	switch s := encoding.Uint16(b); s {
 	case 0, 1, 2:
 		return api.StatusA, nil
-	case 3:
-		if wb.isBasic {
-			return api.StatusA, nil
-		}
-		return api.StatusB, nil
-	case 5, 6, 7, 9:
-		return api.StatusB, nil
+	case 3: // not used correctly by protocol conversion
+		return api.StatusA, nil
```

Comment:
case 3 sollte StatusB sein, da status OCPP nur noch die smart Wallbox bedient, siehe https://github.com/evcc-io/evcc/issues/15592

Reply:
Ok, und was ist mit 9?

Reply:
9 wäre Error/Fault, der für gewöhnlich dann auf den StatusB übersetzt wird.
Ich denke, dass der nicht auf `StatusB` übersetzt werden sollte. Daher von mir in der Logik in meinem Commit https://github.com/evcc-io/evcc/commit/a57a78355afd24bdefb71c1491de2560a6c8dc47 ignoriert (Schleife defaultet auf Error).

Einen Einfluss auf einen etwaigen OCPP-Betrieb darf das ja nicht haben, da evcc im OCPP-Server-Mode den Code in der `delta.go` komplett ignorieren müsste - korrekt?

Reply:
Wobei...
`Register 1000: Wert 9` ("Faulted") und `Register 1001: Wert 7` ("Charging ended with error (vehicle still connected)") könnte man auch in `StatusF` ("Fzg. angeschlossen:   ja    Laden aktiv: nein    Fehler EVSE oder Abstecken simulieren (CP-Wake-up, -12V)") übersetzen.

Thoughts?

Reply:
Bitte den Status 9 und 7 so lassen, wie es ist. Das scheint kein echter Fehler zu sein. Die Basic hatte Status neun gemeldet auf 1000, wenn das Laden beendet wurde. Es ließ sich dann aber wieder normal starten ohne das Fahrzeug aus und einzustecken.

Reply:
Basic verwendet kein 1000 mehr.

Reply:
Das passt. Der Wert kam bei der Basic auch nicht jedes Mal auf dem Register 1000. Es könnte also sein, dass, wenn wir den Fehler neun jetzt nicht mehr auf B Mappen auch irgendwann die Smart ein Problem damit hat und die Leute unnötiger Weise das Auto an und einstecken müssen. 
ich würde vorschlagen, dass Mapping jetzt genau so wie es jetzt abgebildet ist zu lassen und diesen branch zu mergen. es ist jetzt so abgebildet, wie ich jeweils für die ein und andere Box schon einmal in den alten Versionen funktioniert hat

Reply:
Die Smart wird damit kein Problem haben.
Der Wert 9 (Faulted) sollte ja wirklich nur dann ausgegeben werden, wenn es einen Fehler gibt (hatte ich bislang mit der Smart noch nie bie Modbus-Steuerung).

Den Status 9 hatten wir im **Register 1000** in der Vergangenheit nicht behandelt. Das hat im Switch immer in einem "invalid status" geendet:
```

func (wb *Delta) Status() (api.ChargeStatus, error) {
	b, err := wb.conn.ReadInputRegisters(wb.base+deltaRegEvseState, 1)
	if err != nil {
		return api.StatusNone, err
	}

	// 0: Unavailable
	// 1: Available
	// 2: Occupied
	// 3: Preparing
	// 4: Charging
	// 5: Finishing
	// 6: Suspended EV
	// 7: Suspended EVSE
	// 8: Not ready
	// 9: Faulted
	switch s := encoding.Uint16(b); s {
	case 0, 1:
		return api.StatusA, nil
	case 2, 3, 5, 6, 7:
		return api.StatusB, nil
	case 4:
		return api.StatusC, nil
	default:
		return api.StatusNone, fmt.Errorf("invalid status: %0x", s)
	}
}
```